### PR TITLE
More github actions for Compositions

### PR DESCRIPTION
**Pull request by barney-s**
### Change description

Add more github actions for experiments/compositions:

- verify gomod
- verify go imports

### CODE REVIEW

File: .github/workflows/presubmit-compositions.yaml [hunk 1]
```diff
@@ -23,7 +23,48 @@ on:
     branches: ["master"]
     paths:
       - "experiments/compositions/**"
+
+env:
+  GOWORK: off
 jobs:
+  verify-goimports:
```
Comment (acpana) [inline .github/workflows/presubmit-compositions.yaml:30]: could we rely on the golanci-lint uberlinter to run `goimports`? Can be done as a follow on

File: .github/workflows/presubmit-compositions.yaml [hunk 2]
```diff
@@ -23,7 +23,48 @@ on:
...
+    steps:
+      - name: Set up go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.22'
+          stable: true
+      - uses: actions/checkout@v4
+      - name: verify goimports
+        run: |
+          files=$(go run golang.org/x/tools/cmd/goimports -format-only -l experiments/compositions/.)
```
Comment (acpana) [inline .github/workflows/presubmit-compositions.yaml:41]: It may be easier to reproduce any issues w this cmd if we throw these into a script file

**Comment by acpana** [comment]
/lgtm
/approve

2 little call outs; can probably add them as a follow ons if you agree

/hold if you want to address them here.

File: .github/workflows/presubmit-compositions.yaml (hunk 1 above)
Comment (barney-s) [reply .github/workflows/presubmit-compositions.yaml:30]: Will do. We have one more CL in progress: 
https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2493

The top level linter is not catching anything in compositions.

**Comment by barney-s** [comment]
> /lgtm /approve
> 
> 2 little call outs; can probably add them as a follow ons if you agree
> 
> /hold if you want to address them here.

Addressed. Please LGTM again.

File: .github/workflows/presubmit-compositions.yaml [hunk 3]
```diff
@@ -23,7 +23,42 @@ on:
...
+      - name: Set up go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.22'
+          stable: true
+      - uses: actions/checkout@v4
+      - run: |
+          ./experiments/compositions/scripts/github-actions/format-gomod.sh
+      - run: |
+          changes=$(git status --porcelain)
```
Comment (justinsb) [inline .github/workflows/presubmit-compositions.yaml:54]: Tip: I personally prefer to have the GitHub actions be 1:1 with scripts - I think if I was doing it today I would use `dev/ci/presubmits` and auto-generate the GitHub actions.  The advantage of doing so is that then it's easy to run the same thing as GitHub actions locally ... just run the script!

**Review by justinsb — APPROVED** [review]

File: .github/workflows/presubmit-compositions.yaml (hunk 3 above)
Comment (barney-s) [reply .github/workflows/presubmit-compositions.yaml:54]: happy to change it in a follow up.
Q: Is there an example i can look at. ?

**Pull request merged**